ext/re2/re2.cc

```diff
@@ -1358,9 +1364,11 @@ static VALUE re2_regexp_match(int argc, VALUE *argv, const VALUE self) {
     VALUE matchdata = rb_class_new_instance(0, 0, re2_cMatchData);
     TypedData_Get_Struct(matchdata, re2_matchdata, &re2_matchdata_data_type, m);
     m->matches = new(std::nothrow) re2::StringPiece[n];
-    m->regexp = self;
-    m->text = rb_str_dup(text);
-    rb_str_freeze(m->text);
+    RB_OBJ_WRITE(matchdata, &m->regexp, self);
+    if (!RTEST(rb_obj_frozen_p(text))) {
```

Comment:
If we don’t have a test for this already, we should add coverage for passing both a frozen and unfrozen string to `match`.

Reply:
Added in 6cc08657ebdc4b03e5ba03590dc34ab6d5f6579d.

Reply:
Yeah sorry, shouldn't have been in the PR at all, sometimes I just can't help myself to add code to avoid copies :)